src/ui/registries/ViewRegistry.js

```diff
@@ -39,10 +39,12 @@ define(['EventEmitter'], function (EventEmitter) {
     /**
      * @private for platform-internal use
      * @param {*} item the object to be viewed
+     * @param {array} objectPath - The current contextual object path of the view object
+     *                             eg current domainObject is located under MyItems which is under Root
      * @returns {module:openmct.ViewProvider[]} any providers
      *          which can provide views of this object
      */
-    ViewRegistry.prototype.get = function (item) {
+    ViewRegistry.prototype.get = function (item, objectPath = []) {
```

Comment:
I don't think we should default this to an empty array. We should decide whether this is optional or not. If it's optional it should be `undefined`.

I think we want this to be mandatory when requesting a view (because some `canView` functions require it). It is optional for `canView`, `canEdit` etc. to use it, but so long as any of them use it, it must be mandatory to provide it. So, we should probably throw an error if `.get()` is called without an item or an `objectPath`.

Reply:
Subobjectview doesn't call openmct.objectViews.get anywhere.

Reply:
Preview.vue is in this PR: https://github.com/nasa/openmct/pull/3725/files#diff-68f73c796f45cb5038b5afb18e05563fc990b7d45808ad7b00816e9d63b446c2

Reply:
OK, I'm going to shut up now 😄 Sorry. A find on the page didn't show Preview.vue. I mustn't have been showing all commits.

This is clearly going to be one of those days...

Reply:
Let's just throw an error if an `objectPath` isn't provided and then call it a day.